### PR TITLE
Add tests exposing BatchNormalization bug(s)? when used in GANs.

### PR DESCRIPTION
**Pull request by mjdietzx**
Reproducing:

- https://github.com/fchollet/keras/issues/5644
- https://github.com/fchollet/keras/issues/5644

Both tests fail due to bug. They will pass if `batch_normalization` flag is set to `False` in `generator` and `discriminator` for their respective tests.

### CODE REVIEW

**Comment by mjdietzx** [comment]
mean to make this PR to `keras-2`
